Strings_Homework3.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+original_str = '''homEwork:
+
+  tHis iz your homeWork, copy these Text to variable.
+
+
+
+  You NEED TO normalize it fROM letter CASEs point oF View. also, create one MORE senTENCE witH LAST WoRDS of each existING SENtence and add it to the END OF this Paragraph.
+
+
+
+  it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE.
+
+
+
+  last iz TO calculate nuMber OF Whitespace characteRS in this Tex. caREFULL, not only Spaces, but ALL whitespaces. I got 87.
+  '''
+
+sentences = re.split(r'[.?!]', original_str) # create a list consisting of sentences from original string
+last_words = ' '.join([sentences[i].strip().split()[-1] for i in range(len(sentences)-1)]) # create list of last elements (=words) of each sentence and convert this list to string separated by spaces
+normalized_str = '' # declare empty string: we'll add normalized text from original string here
+
+for i in range(len(sentences)): # take each sentence
+    if 'add it to the end of this paragraph' in sentences[i].lower(): # check if there is 'add it to this paragraph' phrase in the sentence
+        sentences.insert(i+1, ' ' + last_words) # if there is, insert 'last words' string into the next element of the list with sentences
+    first_word_match = re.compile(r'(?<=\s)\w+').search(sentences[i]) # in each sentence, find the first word preceded by space.  We need this to separate 'words' in sentence from preceding whitespaces (like multiple newlines in most cases)
+    sliced_sentences = [sentences[i][:first_word_match.start()], sentences[i][first_word_match.start():]] # slice each sentence into 'words part' and 'everything else'
```

Comment:
It works well for now. But we receive the `'NoneType' object has no attribute 'start'` error if we decide to comment the previous "if" statement

Reply:
I'm not sure i understand this concern. What if statement do you mean? On line 25?

Reply:
Yes, on line 25.

Reply:
If you comment if statement, then for loop should be changed to "for i in range(len() - 1) - minus 1 needs to be ended, as this block expects 1 more element to be added (= sentence of last words).